src/core/tests/pass/serialization/deterministicity.cpp

```diff
@@ -91,6 +92,23 @@ TEST_F(SerializationDeterministicityTest, ModelWithMultipleLayers) {
     ASSERT_TRUE(files_equal(bin_1, bin_2));
 }
 
+TEST_F(SerializationDeterministicityTest, Hash) {
+    const std::string model =
+        CommonTestUtils::getModelFromTestModelZoo(ov::util::path_join({SERIALIZED_ZOO, "ir/addmul_abc.onnx"}));
```

Comment:
what specifics we have in onnx files?
can it be reproduced without ONNX file itself? (because ngraph tests on common functionality should not depend on ONNX files)

Reply:
Agree with Ilya. This test can be a part of ONNX frontend tests.
But here can we generate the model in the runtime?

Reply:
Actually, I'm not sure this issue is ONNX specific or not.
Let me summarize my findings here.
I was able to reproduce this issue only with a specific ONNX model on Windows.
With the model, cache blobs were regenerated every new run.
(1. cmd: ./benchmark_app -m [PATH_TO_MODEL] -d GPU -b 1 -hint none -nstreams 1 -niter 1 --cache_dir ./myCache/)
(2. If I run it with the `--load_from_file` option, this issue was not reproduced.)
I dumped the input of `ModelCache::compute_hash()` into files and found that 'xml' and 'bin' files were changed every new run.
In this case, some uninitialized buffers made different memory objects and it prevented consistent constant compression of nGraph serializer. Due to this reason, 'xml' and 'bin' files were changed every new run.
So, I updatedd `ngraph_malloc` to use std::calloc instead of std::malloc.
I don't have a deep understanding on this part, please guide me to the right direction. Thank you.

Reply:
what if you convert this model to IR first and then use caching?

Looks like ONNX frontend has uninitialized memory for some constants (which is strange). Could you please provide model name ? @mbencer could you please comment please?

Reply:
I failed to convert it to IR with mo.
For the model name, please find it in the JIRA ticket. (I'm not sure if it's okay to mention here.)

Reply:
Looks like calloc just hide the sporadic problem. It is better to rootcause